enterprise/server/remote_execution/containers/firecracker/firecracker.go

```diff
@@ -340,25 +339,16 @@ type FirecrackerContainer struct {
 	cancelVmCtx context.CancelFunc
 }
 
-// ConfigurationHash returns a digest that can be used to look up or save a
-// cached snapshot for this container configuration.
-func (c *FirecrackerContainer) ConfigurationHash() *repb.Digest {
-	params := []string{
-		fmt.Sprintf("cpus=%d", c.constants.NumCPUs),
-		fmt.Sprintf("mb=%d", c.constants.MemSizeMB),
-		fmt.Sprintf("scratch=%d", c.constants.ScratchDiskSizeMB),
-		fmt.Sprintf("net=%t", c.constants.EnableNetworking),
-		fmt.Sprintf("dockerd=%t", c.constants.InitDockerd),
-		fmt.Sprintf("debug=%t", c.constants.DebugMode),
-		fmt.Sprintf("container=%s", c.containerImage),
-	}
```

Comment:
these were effectively invariants -- because they cannot be changed between vm runs from a snapshot. Are you sure you want to remove these rather than incorporating them into the key?

Reply:
Yeah I could imagine introducing a default that we don't allow specifying via a platform prop that is also an invariant.

I like your suggestion of hashing the constants struct and including it in the action salt.